crates/kubelet/src/state.rs

```diff
@@ -58,8 +62,8 @@ impl<PodState> Transition<PodState> {
     ///         self: Box<Self>,
     ///         _pod_state: &mut PodState,
     ///         _pod: &Pod,
-    ///     ) -> anyhow::Result<Transition<PodState>> {
-    ///         Ok(Transition::next(self, TestState))
+    ///     ) -> Transition<PodState> {
```

Comment:
One of the drawbacks to our `compile_fail` doctests are that they need to be chanced here too, but pass because they still fail to compile with this change. I wish you could match on a particular compile error.

Reply:
Good catch - thanks!